src/api/db/migrate/20180405074538_change_default_charset_and_mysql57_defaults.rb

```diff
@@ -0,0 +1,45 @@
+class ChangeDefaultCharsetAndMysql57Defaults < ActiveRecord::Migration[5.1]
+  # mysql/mariadb has changed the default to utf8mb4, we need to update old instances
+  # to avoid to have specific sql code for difference mysql versions
+  def up
+    # these may be utf8 or utf8mb4 depending on the use mysql instance on creation time
+    [ 'ar_internal_metadata', 'binary_releases', 'bs_request_counter',
+      'cloud_azure_configurations', 'cloud_ec2_configurations', 
+      'cloud_user_upload_jobs', 'data_migrations', 'download_repositories',
+      'group_maintainers', 'incident_updateinfo_counter_values',
+      'kiwi_images', 'kiwi_package_groups', 'kiwi_packages',
+      'kiwi_preferences', 'kiwi_repositories', 'maintained_projects',
+      'product_media', 'product_update_repositories' ].each do |table|
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8");
+    end
+    # contains free text, could have emojis
+    [ 'configurations', 'history_elements', 'kiwi_descriptions', 'notifications' ].each do |table|
+      execute("ALTER TABLE #{table} ROW_FORMAT=DYNAMIC"); # default in MySQL 5.7 & MariaDB 10.2
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8mb4");
+    end
+    execute('alter table cloud_azure_configurations modify application_id mediumtext DEFAULT NULL;')
+    execute('alter table cloud_azure_configurations modify application_key mediumtext DEFAULT NULL;')
+    execute('alter table download_repositories modify pubkey mediumtext DEFAULT NULL;')
+    execute('alter table history_elements modify comment mediumtext DEFAULT NULL;')
+    execute('alter table notifications modify event_payload mediumtext NOT NULL;')
+  end
+
+  def down
+    # just normalize, it depends on which mysql instance the tables got created, so we can not
+    # really restore it
+    [ 'ar_internal_metadata', 'binary_releases', 'bs_request_counter',
+      'cloud_azure_configurations', 'cloud_ec2_configurations', 
+      'cloud_user_upload_jobs', 'configurations', 'data_migrations', 'download_repositories',
+      'group_maintainers', 'history_elements', 'incident_updateinfo_counter_values',
+      'kiwi_descriptions', 'kiwi_images', 'kiwi_package_groups', 'kiwi_packages',
+      'kiwi_preferences', 'kiwi_repositories', 'maintained_projects',
+      'notifications', 'product_media', 'product_update_repositories' ].each do |table|
+      execute("ALTER TABLE #{table} CONVERT TO CHARACTER SET utf8");
+    end
+    execute('alter table cloud_azure_configurations modify application_id text DEFAULT NULL;')
+    execute('alter table cloud_azure_configurations modify application_key text DEFAULT NULL;')
+    execute('alter table download_repositories modify pubkey text DEFAULT NULL;')
+    execute('alter table history_elements modify comment text DEFAULT NULL;')
+    execute('alter table notifications modify event_payload text NOT NULL;')
```

Comment:
Same here basically. Have you tried this?

Reply:
yes, this code did run against our production data at least. I have not used that instance though ...

Reply:
Okay then question is what did it do? Just cut off the text?

Reply:
one more commit to fix this mismatch

Reply:
nah, seems a former run set it to mediumtext ... cleaned this up now, we don't need it hopefully.